layouts/default.vue

```diff
@@ -112,45 +112,96 @@
 </template>
 
 <script lang="ts">
+import { stringify } from 'qs';
 import Vue from 'vue';
 import { mapActions, mapState } from 'vuex';
 
+interface WebSocketMessage {
+  MessageType: string;
+  Data?: never;
+}
+
 export default Vue.extend({
   data() {
     return {
       drawer: true,
-      opacity: 0
+      opacity: 0,
+      keepAliveInterval: undefined as number | undefined
     };
   },
   computed: {
-    ...mapState('page', ['opaqueAppBar']),
-    items() {
-      return [
-        {
-          icon: 'mdi-home',
-          title: this.$t('home'),
-          to: '/'
-        }
-      ];
+    items: {
+      get(): Array<Record<string, string>> {
+        return [
+          {
+            icon: 'mdi-home',
+            title: this.$t('home') as string,
+            to: '/'
+          }
+        ];
+      }
     },
-    libraries() {
-      return this.$store.getters['userViews/getNavigationDrawerItems'];
+    libraries: {
+      get() {
+        return this.$store.getters['userViews/getNavigationDrawerItems'];
+      }
     },
-    configItems() {
-      return [
-        {
-          icon: 'mdi-cog',
-          title: this.$t('settings'),
-          to: '/settings'
-        }
-      ];
+    configItems: {
+      get(): Array<Record<string, string>> {
+        return [
+          {
+            icon: 'mdi-cog',
+            title: this.$t('settings') as string,
+            to: '/settings'
+          }
+        ];
+      }
     }
   },
   beforeMount() {
     this.refreshUserViews();
+
+    const socketParams = stringify({
+      api_key: this.$store.state.user.accessToken,
+      deviceId: this.$store.state.deviceProfile.deviceId
+    });
+    let socketUrl = `${this.$axios.defaults.baseURL}/socket?${socketParams}`;
+    socketUrl = socketUrl.replace('https:', 'wss:');
+    socketUrl = socketUrl.replace('http:', 'ws:');
+
+    this.$connect(socketUrl);
+    this.handleKeepAlive();
+  },
+  beforeDestroy() {
+    if (this.keepAliveInterval) {
+      clearInterval(this.keepAliveInterval);
+    }
   },
   methods: {
-    ...mapActions('userViews', ['refreshUserViews'])
+    ...mapActions('userViews', ['refreshUserViews']),
+    ...mapState('page', ['opaqueAppBar']),
+    handleKeepAlive() {
+      this.$store.subscribe((mutation, state) => {
+        if (
+          mutation.type === 'SOCKET_ONMESSAGE' &&
+          state.socket.message.MessageType === 'ForceKeepAlive'
+        ) {
+          this.sendWebSocketMessage('KeepAlive');
+          this.keepAliveInterval = setInterval(() => {
+            this.sendWebSocketMessage('KeepAlive');
+          }, state.socket.message.Data * 1000 * 0.5);
+        }
+      });
+    },
+    sendWebSocketMessage(name: string, data: Record<string, never>) {
+      const msg: WebSocketMessage = { MessageType: name };
+
+      if (data) {
+        msg.Data = data;
+      }
+
+      this.$store.state.socket.instance.send(JSON.stringify(msg));
+    }
```

Comment:
If data is optional, I think it'd be easier to send the params as an object, and have data be optional
```
    sendWebSocketMessage({name, data}: {name: string, data?: Record<string, never>}) {

Reply:
Indeed, oversight on my part.